src/test/regress/upgrade/config.py

```diff
@@ -1,68 +1,299 @@
-MIXED_BEFORE_CITUS_UPGRADE_SCHEDULE = './mixed_before_citus_upgrade_schedule'
-MIXED_AFTER_CITUS_UPGRADE_SCHEDULE = './mixed_after_citus_upgrade_schedule'
+CUSTOM_SCHEDULE_NAMES = [
+    "custom_create_schedule",
+    "custom_sql_schedule",
+    "custom_postgres_schedule",
+]
 
-MASTER = 'master'
+BEFORE_PG_UPGRADE_SCHEDULE = "./before_pg_upgrade_schedule"
+AFTER_PG_UPGRADE_SCHEDULE = "./after_pg_upgrade_schedule"
+
+CUSTOM_CREATE_SCHEDULE = "./custom_create_schedule"
+CUSTOM_POSTGRES_SCHEDULE = "./custom_postgres_schedule"
+CUSTOM_SQL_SCHEDULE = "./custom_sql_schedule"
+
+AFTER_CITUS_UPGRADE_COORD_SCHEDULE = "./after_citus_upgrade_coord_schedule"
+BEFORE_CITUS_UPGRADE_COORD_SCHEDULE = "./before_citus_upgrade_coord_schedule"
+MIXED_BEFORE_CITUS_UPGRADE_SCHEDULE = "./mixed_before_citus_upgrade_schedule"
+MIXED_AFTER_CITUS_UPGRADE_SCHEDULE = "./mixed_after_citus_upgrade_schedule"
+
+CITUS_CUSTOM_TEST_DIR = "./tmp_citus_test"
+
+MASTER = "master"
 # This should be updated when citus version changes
-MASTER_VERSION = '10.2'
+MASTER_VERSION = "10.2"
 
 HOME = expanduser("~")
 
 
 CITUS_VERSION_SQL = "SELECT extversion FROM pg_extension WHERE extname = 'citus';"
 
 
-class CitusUpgradeConfig():
+def find_free_port():
+    with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as s:
+        s.bind(("", 0))
+        s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        return s.getsockname()[1]
+
+
+class NewInitCaller(type):
+    def __call__(cls, *args, **kwargs):
+        obj = type.__call__(cls, *args, **kwargs)
+        obj.init()
+        return obj
 
+
+class CitusBaseClusterConfig(object, metaclass=NewInitCaller):
     def __init__(self, arguments):
-        self.bindir = arguments['--bindir']
-        self.pre_tar_path = arguments['--citus-pre-tar']
-        self.post_tar_path = arguments['--citus-post-tar']
-        self.pg_srcdir = arguments['--pgxsdir']
-        self.temp_dir = './tmp_citus_upgrade'
-        self.datadir = self.temp_dir + '/data'
+        if "--bindir" in arguments:
+            self.bindir = arguments["--bindir"]
+        self.pg_srcdir = arguments["--pgxsdir"]
+        self.temp_dir = CITUS_CUSTOM_TEST_DIR
+        self.worker_amount = 2
+        self.user = REGULAR_USER_NAME
+        self.is_mx = False
+        self.is_citus = True
+        self.name = type(self).__name__
         self.settings = {
-            'shared_preload_libraries': 'citus',
-            'citus.node_conninfo': 'sslmode=prefer',
-            'citus.enable_version_checks' : 'false'
+            "shared_preload_libraries": "citus",
+            "citus.node_conninfo": "sslmode=prefer",
+            "citus.enable_repartition_joins": True,
+            "citus.repartition_join_bucket_count_per_node": 2,
+            "max_connections": 600,
         }
-        self.mixed_mode = arguments['--mixed']
+        self.new_settings = {}
+        self.add_coordinator_to_metadata = False
```

Comment:
This prevents use of citus local tables and hence any operation with local tables. Can we set it to true do you think?

Reply:
I know that this is an important setting to have both true/false in the tests. But, given HSC does this and interactions with local tables are crucial, I think we should make True by default. What is your take?